File: src/acceptance-test/java/net/slipp/support/SharedDriver.java
package net.slipp.support;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.events.EventFiringWebDriver;

public class SharedDriver extends EventFiringWebDriver {
    private static final WebDriver REAL_DRIVER = WebDriverFactory.createWebDriver();
    private static final Thread CLOSE_THREAD = new Thread() {
        @Override
        public void run() {
            REAL_DRIVER.quit();
        }
    };
 
    static {
        Runtime.getRuntime().addShutdownHook(CLOSE_THREAD);
    }
 
    public SharedDriver() {
        super(REAL_DRIVER);
    }
 
    @Override
    public void close() {
        if(Thread.currentThread() != CLOSE_THREAD) {
            throw new UnsupportedOperationException("You shouldn't close this WebDriver. It's shared and will close when the JVM exits.");
        }
        super.quit();
    }
 
    public void deleteAllCookies() {
        manage().deleteAllCookies();
    }
}

File: src/acceptance-test/scala/net/slipp/support/SharedDriver.scala
package net.slipp.support

import org.openqa.selenium.WebDriver
import org.openqa.selenium.support.events.EventFiringWebDriver

object SharedDriver {
  private val REAL_DRIVER: WebDriver = WebDriverFactory.createWebDriver
  private val CLOSE_THREAD: Thread = new Thread() {
    override def run {
      REAL_DRIVER.quit
    }
  }

  Runtime.getRuntime.addShutdownHook(CLOSE_THREAD)
}

class SharedDriver extends EventFiringWebDriver(SharedDriver.REAL_DRIVER) {
  override def close {
    if (Thread.currentThread ne SharedDriver.CLOSE_THREAD) {
      throw new UnsupportedOperationException("You shouldn't close this WebDriver. It's shared and will close when the JVM exits.")
    }
    super.quit
  }

  def deleteAllCookies {
    manage.deleteAllCookies
  }
}
